src/test/java/Homework22.java

```diff
@@ -0,0 +1,36 @@
+import org.openqa.selenium.WebDriver;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import pages.BasePage;
+import pages.HomePage;
+import pages.LoginPage;
+
+
+
+
+public class Homework22 extends BasePage
+{
+
+    public Homework22(WebDriver givenDriver)
+    {
+        super(givenDriver);
+    }
+    public class PlaylistTests
+    {
+        @Test(dataProvider = "CorrectLoginProvider", dataProviderClass = BaseTest.class)
+        public void renamePlaylist(String email, String password) {
```

Comment:
Hi friend,
I have checked your homework 22, but test script Homework22 can't run. 

// HINT:
1. You need to update `public class Homework22 entends BaseTest` not is BasePage
-> because in BaseTest class define method to setup and initialize driver. So the test scripts just need to extend BaseTest will inherit all the properties of the parent BaseTest class -> not need to declare WebDriver driver anymore.
2. Delete constructor `Homework22`
3.  Delete` public class PlaylistTest `
You cant paste code below and try run test script again

```
public class Homework22 extends BaseTest
{
        @Test(dataProvider = "CorrectLoginProvider", dataProviderClass = BaseTest.class)
        public void renamePlaylist(String email, String password) {

            String playlistName = "My test playlist";

            LoginPage loginPage = new LoginPage(driver);
            HomePage homePage = new HomePage(driver);

            loginPage.login();

            homePage.doubleClickPlaylist();
            homePage.enterNewPlaylistName();

            Assert.assertTrue(homePage.doesPlaylistNameExist(playlistName));
        }
}
```